main.py

```diff
@@ -0,0 +1,136 @@
+# pylint: disable=superfluous-parens
+import os
+from typing import Dict, Tuple
+from uuid import UUID
+
+import simplejson as json
+from flask import Request
+from notifications_python_client import __version__
+from notifications_python_client.authentication import create_jwt_token
+from requests import Session
+from requests.exceptions import RequestException
+
+session = Session()
+
+
+NOTIFY_API_KEY = os.environ["NOTIFY_API_KEY"]
```

Comment:
There is a scenario whereby the key could also be deleted in the function's environment, so this would still need to raise a `KeyError` in that scenario. Given that, I will keep this line as a dictionary key access, and add something to the deployment script to test it before we get to run time

Reply:
Also, I'm not sure we have fully considered what we want to do with the notify key, e.g. should it be provisioned via secrets manager?

Reply:
Not yet, but as discussed, we could either store the key in the k8s secrets for each env (wouldn't have to conditional fetch values 🤔 ) or have a secret's manager for each env.

Reply:
I've added an early exit to the deploy script if this isn't set